### DISCUSSION TITLE
How do I return the value of Input() ?

### DISCUSSION THREAD

**Discussion by totordev** [Textualize/textual#2384] [Q&A] [answered]
Hello, I am shebangcode and I'm trying to create a TUI for my app, but I don't know how to return the value of Input(). Alas, I need to know how to do it to fulfill my needs. Can someone help me?

**Comment by davep** [comment, marked as answer]
Pretty much the same pattern as outlined in https://github.com/Textualize/textual/discussions/2376 -- only this time you'll be using the [`Submitted`](https://textual.textualize.io/widgets/input/#textual.widgets._input.Input.Submitted) message from the `Input`.

**Reply by totordev** [reply]
Yes, now I understand, but how can I return the text that the user puts in Input()?

**Reply by totordev** [reply]
Oh! Nevermind, I've just read the documentation and I understand it completely now. Thanks!

**Reply by davep** [reply]
In the documentation I just linked to you should see this:

<img width="662" alt="Screenshot 2023-04-26 at 14 41 39" src="https://user-images.githubusercontent.com/28237/234594230-5a265ed3-d71d-4aab-98f3-92bfe8631a99.png">

**Reply by totordev** [reply]
Yes, I've just read it.
